x/evm/precompile/wasm.go

```diff
@@ -38,11 +38,14 @@ func (p precompileWasm) Run(
 	defer func() {
 		err = ErrPrecompileRun(err, p)
 	}()
-	startResult, err := OnRunStart(evm, contract.Input, p.ABI())
+	startResult, err := OnRunStart(evm, contract.Input, p.ABI(), contract.Gas)
 	if err != nil {
 		return nil, err
 	}
 
+	// Resets the gas meter to parent one after precompile execution and gracefully handles "out of gas"
+	defer HandleOutOfGasPanic(&err)()
```

Comment:
Look like this could be inside `OnRunStart` so that you won't need to write it in multiple places